providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolverTest.java

```diff
@@ -0,0 +1,267 @@
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.StorageState;
+import dev.openfeature.sdk.ErrorCode;
+import dev.openfeature.sdk.ImmutableContext;
+import dev.openfeature.sdk.MutableContext;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.ProviderState;
+import dev.openfeature.sdk.Reason;
+import dev.openfeature.sdk.Value;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Field;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import static dev.openfeature.contrib.providers.flagd.resolver.process.MockFlags.BOOLEAN_FLAG;
+import static dev.openfeature.contrib.providers.flagd.resolver.process.MockFlags.DISABLED_FLAG;
+import static dev.openfeature.contrib.providers.flagd.resolver.process.MockFlags.FLAG_WIH_IF_IN_TARGET;
+import static dev.openfeature.contrib.providers.flagd.resolver.process.MockFlags.FLAG_WIH_INVALID_TARGET;
+import static dev.openfeature.contrib.providers.flagd.resolver.process.MockFlags.OBJECT_FLAG;
+import static dev.openfeature.contrib.providers.flagd.resolver.process.MockFlags.VARIANT_MISMATCH_FLAG;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTimeout;
+import static org.junit.jupiter.api.Assertions.assertTimeoutPreemptively;
+
+class InProcessResolverTest {
+
+    @Test
+    public void eventHandling() throws Throwable {
+        // given
+        // note - queues with adequate capacity
+        final BlockingQueue<StorageState> sender = new LinkedBlockingQueue<>(5);
+        final BlockingQueue<ProviderState> receiver = new LinkedBlockingQueue<>(5);
+
+        InProcessResolver inProcessResolver =
+                getInProcessResolverWth(new MockStorage(new HashMap<>(), sender), providerState -> {
+                    receiver.offer(providerState);
+                });
+
+        // when
+        inProcessResolver.init();
+        if (!sender.offer(StorageState.OK, 200, TimeUnit.MILLISECONDS)) {
+            Assertions.fail("failed to send the event");
+        }
+
+        if (!sender.offer(StorageState.ERROR, 200, TimeUnit.MILLISECONDS)) {
+            Assertions.fail("failed to send the event");
+        }
+
+        // then
+        assertTimeoutPreemptively(Duration.ofMillis(200), () -> {
+            Assertions.assertEquals(ProviderState.READY, receiver.take());
+        });
+
+        assertTimeoutPreemptively(Duration.ofMillis(200), () -> {
+            Assertions.assertEquals(ProviderState.ERROR, receiver.take());
+        });
+    }
+
+    @Test
+    public void simpleResolving() throws Exception {
```

Comment:
maybe worth testing also for numeric values: int and double

Reply:
Added them in the latest commit :)